symbolic-base/src/ZkFold/FFI/Rust/Conversion.hs

```diff
@@ -49,6 +53,41 @@ instance {-# OVERLAPPABLE #-} (RustHaskell r h, Storable h) => Storable r where
 --  pokeElemOff = error "Do not call pokeElemOff on Rust type"
 --  pokeByteOff = error "Do not call pokeByteOff on Rust type"
 
+pointSize :: Int
+pointSize = sizeOf (undefined :: Rust_BLS12_381_G1_Point)
+
+scalarSize :: Int
+scalarSize = sizeOf (undefined :: Fr)
+
+o2nScalar :: Fr -> Fr
+o2nScalar old = unsafePerformIO $ do
+    withForeignPtr (rawData $ rawScalar old) $ \ptr -> do
+        ptrNew <- r_h2r_scalar ptr scalarSize
+        RScalar . RData <$> (newForeignPtr finalizerFree ptrNew )
```

Comment:
I think we need to run a finalizer which is different from standard `finalizerFree`, I don't know if this would work with Rust types